src/plugins/proposal.ts

```diff
@@ -0,0 +1,688 @@
+    type?: ProposalName;
+    [key: string]: any | undefined;
+  }
+}
+
+export interface IProposalBaseCreateOptions {
+  dao: Address
+  description?: string
+  descriptionHash?: string
+  title?: string
+  tags?: string[]
+  plugin?: Address
+  url?: string
+}
+
+export interface IProposalState {
+  // TODO: semantically order props
+  id: string
+  dao: IEntityRef<DAO>
+  votingMachine: Address
+  // TODO: plugin instance inside itself? or other type of plugin?
+  plugin: IEntityRef<AnyPlugin>
+  closingAt: number
+  createdAt: number | Date
+  descriptionHash?: string
+  description?: string
+  name: string
+  executedAt: number
+  organizationId: string
+  paramsHash: string
+  // TODO: Stores proposal instance inside itself? Or is this another proposaltype?
+  proposal: IEntityRef<AnyProposal>
+  proposer: Address
+  resolvedAt: number
+  tags?: string[]
+  title?: string
+  totalRepWhenCreated: BN
+  totalRepWhenExecuted: BN
+  type: ProposalName
+  url?: string
+  votes: Array<IEntityRef<Vote>>
+  votesFor: BN
+  votesAgainst: BN
+  votesCount: number
+  voteOnBehalf: Address
+  winningOutcome: IProposalOutcome
+
+  // Genesis Protocol Proposal
+  queue: IEntityRef<Queue>
+  quietEndingPeriodBeganAt: number
+  stage: IProposalStage
+  accountsWithUnclaimedRewards: Address[]
+  boostedAt: number
+  upstakeNeededToPreBoost: BN
+  stakesFor: BN
+  stakesAgainst: BN
+  preBoostedAt: number
+  genesisProtocolParams: IGenesisProtocolParams
+  executionState: IExecutionState
+  expiresInQueueAt: number
+  downStakeNeededToQueue: BN
+  confidenceThreshold: number
+}
+
+export abstract class Proposal<TProposalState extends IProposalState> extends Entity<
+  TProposalState
+> {
+  public static get baseFragment(): DocumentNode {
+    if (!this.baseFragmentField) {
+      this.baseFragmentField = gql`fragment ProposalFields on Proposal {
+        id
+        accountsWithUnclaimedRewards
+        boostedAt
+        closingAt
+        confidenceThreshold
+        createdAt
+        dao {
+          id
+          schemes {
+            id
+            address
+          }
+        }
+        description
+        descriptionHash
+        executedAt
+        executionState
+        expiresInQueueAt
+        genesisProtocolParams {
+          id
+          activationTime
+          boostedVotePeriodLimit
+          daoBountyConst
+          limitExponentValue
+          minimumDaoBounty
+          preBoostedVotePeriodLimit
+          proposingRepReward
+          queuedVotePeriodLimit
+          queuedVoteRequiredPercentage
+          quietEndingPeriod
+          thresholdConst
+          votersReputationLossRatio
+        }
+        gpRewards {
+          id
+        }
+        scheme {
+          ...PluginFields
+        }
+        gpQueue {
+          id
+          threshold
+          votingMachine
+        }
+        organizationId
+        preBoostedAt
+        proposer
+        quietEndingPeriodBeganAt
+
+        stage
+        # stakes { id }
```

Comment:
commenting this out?

Reply:
`stakes` field was already commented and was not being used. Should I remove it completely?

Reply:
I'm not sure, @jellegerbrandy?